search/search_repository.go

```diff
@@ -239,6 +245,158 @@ func parseSearchString(rawSearchString string) (searchKeyword, error) {
 	return res, nil
 }
 
+func parseMap(queryMap map[string]interface{}, q *Query) {
+	for key, val := range queryMap {
+		switch concreteVal := val.(type) {
+		case []interface{}:
+			q.Name = key
+			parseArray(val.([]interface{}), &q.Children)
+		case string:
+			q.Name = key
+			s := string(concreteVal)
+			q.Value = &s
+		case bool:
+			s := concreteVal
+			q.Negate = s
+		}
```

Comment:
Should we consider any other type ?  Should we log something if it is that unsupported type ?